extension.ts

```diff
@@ -227,6 +232,54 @@ async function createWebApp(context?: AzureImageTagNode | DockerHubImageTagNode)
   } else if (result.status === 'Cancelled') {
     throw new UserCancelledError();
   }
+  const website = wizard.createdWebSite;
+  let appUri = `https://${website.name}.scm.azurewebsites.net/docker/hook`;
+
+  if (context instanceof AzureImageTagNode) {
+    await createWebhookForWebApp(context, wizard, appUri);
+
+  } else {
+    // point to dockerhub to create a webhook
+    // http://cloud.docker.com/repository/docker/<registryName>/<repoName>/webHooks
+    const dockerhubPrompt: string = "Copy webapp endpoint and head to dockerhub";
+    let response: string = await vscode.window.showInformationMessage("Please head to your dockerhub account to set up a CI/CD webhook", dockerhubPrompt);
+    if (response) {
+      // tslint:disable-next-line:no-unsafe-any
+      clipboardy.writeSync(appUri);
+      // tslint:disable-next-line:no-unsafe-any
+      opn(`https://cloud.docker.com/repository/docker/${context.userName}/${context.repositoryName}/webHooks`);
+    }
+  }
+  return;
+}
+
+async function createWebhookForWebApp(context: AzureImageTagNode, wizard: WebAppCreator, appUri: string): Promise<void> {
+  //create webhook
+  const crmClient: ContainerRegistryManagementClient = await AzureUtilityManager.getInstance().getContainerRegistryManagementClient(context.subscription);
+  let resourceGroup: ResourceGroup = (<ResourceGroupStep>wizard.findStep(step => step instanceof ResourceGroupStep, "Resource Group step not executed")).resourceGroup;
+  let websiteStep: WebsiteStep = (<WebsiteStep>wizard.findStep(step => step instanceof WebsiteStep, ""));
+  let registryName: string = websiteStep.registry.name;
+  let webhookName: string = `webapp${websiteStep.website.name}`;
+
+  //verify that the appropriate webhook doesn't already exist
+  if ((await crmClient.webhooks.list(resourceGroup.name, registryName)).find((hook) => hook.name === webhookName)) {
+    return;
```

Comment:
Should users specify the hook name?  Can there easily be conflicts?

Reply:
Users don't specify the name. The webhook name is dependent on the name of the web app. 
Regarding conflicts, I assume that a user creating a webhook that's precisely named to conflict with the web-app-CI webhook would be fairly improbable.